src/frequenz/channels/experimental/_pipe.py

```diff
@@ -0,0 +1,85 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Pipe between a receiver and a sender.
+
+The `Pipe` class takes a receiver and a sender and creates a pipe between them by
+forwarding all the messages received by the receiver to the sender.
+"""
+
+from __future__ import annotations
+
+import asyncio
+import typing
+
+from .._generic import ChannelMessageT
+from .._receiver import Receiver
+from .._sender import Sender
+
+
+class Pipe(typing.Generic[ChannelMessageT]):
+    """A pipe between two channels.
+
+    The `Pipe` class takes a receiver and a sender and creates a pipe between them
+    by forwarding all the messages received by the receiver to the sender.
+
+    Example:
+        ```python
+        from frequenz.channels import Broadcast, Pipe
+
+        channel1: Broadcast[int] = Broadcast(name="channel1")
+        channel2: Broadcast[int] = Broadcast(name="channel2")
+
+        async with Pipe(channel2.new_receiver(), channel1.new_sender()):
+            sender_chan2 = channel2.new_sender()
+            receiver_chan1 = channel1.new_receiver()
```

Comment:
This could be also created outside the `async with`, right? It got me a bit confused because somehow I was expecting that `Pipe` was an object that you use more actively, like `pipe.send()` and `pipe.receive()`, like the `RelaySender`. I guess it doesn't make sense to implement it like that, but that's what I had in mind while talking about it for some reason :facepalm: 

I would move it outside to make it more clear.

----

Probably not for this PR, more like brainstorming:

With this in mind, and if this is mostly intended to be used as a context manager, then maybe we could create a function called `connecting()` that creates the `Pipe` (name inspired by [closing()](https://docs.python.org/3/library/contextlib.html#contextlib.closing)) context manger.

```py
async with connecting(channel2.new_receiver(), channel1.new_sender()):
    await sender_chan2.send(42)
    assert await receiver_chan1.receive() == 42
```

(it could be `piping()` too but IMHO it sounds weirder as a verb, and `connecting` seems clear enough).

Also, I understand this is useful as it is, and it does what it says it does, but for me it is a bit confusing that is like having a side effect, you are not using the sender and receiver that you make part of the `Pipe`, is more like connecting channels (I mean the end effect), and this depends on the type of channel. I don't have a proposal for solving this, just mentioning that got me confused for a bit. I thought of an options that makes it more clear to me but I guess it's not suitable: 

Having a `async with connecting(source: Broadcast, destination: Broadcast):`, this for me sounds much more clear, but I guess we probably want to use this in a place where we just have a sender and a receiver, not the channels.

Reply:
> I would move it outside to make it more clear.

done.